tests/main.cpp

```diff
@@ -724,3 +724,49 @@ TEST_CASE("getFirstAvailableBlock") {
   const auto firstAvailableBlock = connection.getFirstAvailableBlock();
   CHECK_GT(firstAvailableBlock, 0);
 }
+
+TEST_CASE("getSlot") {
+  const solana::Keypair keyPair = solana::Keypair::fromFile(KEY_PAIR_FILE);
+  const auto connection = solana::rpc::Connection(solana::DEVNET);
+  const auto slot = connection.getSlot();
+  CHECK_GT(slot, 0);
+  sleep(2);  // sleep for 2 seconds
+  const auto slot2 = connection.getSlot();
+  CHECK_GE(slot2, slot);
+}
+
+TEST_CASE("getSlotLeader") {
+  const solana::Keypair keyPair = solana::Keypair::fromFile(KEY_PAIR_FILE);
+  const auto connection = solana::rpc::Connection(solana::DEVNET);
+  const auto slotLeader = connection.getSlotLeader();
+  boost::regex expression{
+      "^(?!.*[0OlI_])\\w*$"};  //  regex for base-58 encoded string
+  CHECK_EQ(boost::regex_match(slotLeader, expression), true);
+}
+
+TEST_CASE("minimumLedgerSlot") {
+  const solana::Keypair keyPair = solana::Keypair::fromFile(KEY_PAIR_FILE);
+  const auto connection = solana::rpc::Connection(solana::DEVNET);
+  const auto slot = connection.getminimumLedgerSlot();
+  CHECK_GT(slot, 0);
+}
+
+TEST_CASE("getGenesisHash") {
+  const solana::Keypair keyPair = solana::Keypair::fromFile(KEY_PAIR_FILE);
+  const auto connection = solana::rpc::Connection(solana::DEVNET);
+  const auto hash = connection.getGenesisHash();
+  boost::regex expression{
```

Comment:
genesis hash is basically a constant. no need for regex

Reply:
Removed regex check ( now checking that the string should not be empty)

Reply:
please check for the exact hash, each network has a constant hash. it should not change

Reply:
Done